### DISCUSSION TITLE
README contradiction

### DISCUSSION THREAD

**Discussion by mcandre** [xmake-io/xmake#3316] [General]
The README says that Xmake has no dependencies.

Then it says it needs Lua.

Then is says it needs Make/Ninja + CMake/Meson + Vcpkg/Conan + distcc + ccache.

So it has dependencies.

**Comment by waruqi** [comment]
> Then it says it needs Lua.

There are no installation dependencies, it has embed lua vm, you just need to compile and install xmake, only one executable file. You don't need to install lua additionally.

> Then is says it needs Make/Ninja + CMake/Meson + Vcpkg/Conan + distcc + ccache.

no, xmake itself contains the main features of these tools `Make/Ninja + CMake/Meson + Vcpkg/Conan + distcc + ccache`. So you can use xmake and replace them completely.

xmake can build project like Make/Ninja to build projects directly, without relying on make/ninja.

xmake can generate project files such as vcproj/makefile/build.ninja, like Cmake/Meson, without relying on cmake/meson.

xmake has a built-in package manager, like Vcpkg/Conan, providing close to 800+ c++ packages without relying on vcpkg/conan.

xmake has a built-in distributed compilation system like distcc, and also supports windows, which distcc does not.

xmake has built-in local build cache support, like ccache. and it support windows too, and remote caching support, without relying on ccache.
